patroni/dcs/consul.py

```diff
@@ -402,14 +412,23 @@ def touch_member(self, data, permanent=False):
             self._client.kv.delete(self.member_path)
             create_member = True
 
+        if self._register_service or self._previous_loop_register_service:
+            try:
+                self.update_service(not create_member and member and member.data or {}, data)
+            except InvalidSession:
+                self._session = None
+                logger.error('Our session disappeared from Consul, can not "touch_member"')
+                return False
+            except Exception:
+                logger.exception('touch_member')
+                return False
```

Comment:
The `update_service()` isn't supposed to raise exceptions, at least not the `InvalidSession`.

```suggestion
            try:
                self.update_service(not create_member and member and member.data or {}, data)
            except Exception:
                logger.exception('update_service')
```